spark/src/main/scala/geotrellis/spark/RasterMetaData.scala

```diff
@@ -2,26 +2,37 @@ package geotrellis.spark
 
 import geotrellis.raster._
 import geotrellis.spark.tiling._
-import geotrellis.vector.Extent
+import geotrellis.vector.{ProjectedExtent, Extent}
 
 import geotrellis.proj4.CRS
 
 import org.apache.spark.rdd._
 
+/**
+ * @param cellType    value type of each cell
+ * @param layout      definition of the tiled raster layout
+ * @param dataExtent  Extent covering the source data
+ * @param crs         CRS of the raster projection
+ */
 case class RasterMetaData(
   cellType: CellType,
-  extent: Extent,
-  crs: CRS,
-  tileLayout: TileLayout
+  layout: LayoutDefinition,
+  dataExtent: Extent,
```

Comment:
Even with nested encoding distinction between layout extent (extent covering the tiles) and data extent (extent covering the cells with data) is not obvious, and this is something we missed and got wrong the first time around. Maybe doc-string is all that's required, but I felt hurt enough to give it a special name. Given that reasoning is it still too awkward for the benefit ?